### PR TITLE
Setup redux dq6bht fw

### PR DESCRIPTION
**Pull request by mansurissa**
---
#### Description of Task to be completed?
 - Setup redux
 - create  a store for our states
 - creating action, reducer of a counter for testing
 -  using redux-thunk muddleware for handling promises
 - configuring our app to use redux dev tools
 ---
#### How should this be manually tested?
- clone it toyour local machine
- run `npm install` to install all dependencies
- run `npm start` and go to `/counter`
---
#### What are the relevant trello id?
[Dq6bhtFw](https://trello.com/c/Dq6bhtFw/26-setup-redux)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /node_modules
+<<<<<<< HEAD
```
Comment (samuel776) [inline .gitignore]: could you please remove these head-----

File: .gitignore [hunk 2]
```diff
@@ -11,3 +12,7 @@
 dist/
 npm-debug.log*
 .eslintcache
+=======
```
Comment (samuel776) [inline .gitignore]: the same here you can remove that lines

File: .gitignore [hunk 3]
```diff
@@ -11,3 +12,7 @@
...
+.DS_Store
+dist/
+>>>>>>> chore(react):setup React
```
Comment (samuel776) [inline .gitignore]: the same here you can remove commit messages from conflicts solving

File: README.md [hunk 4]
```diff
@@ -11,7 +11,11 @@ Phantom is an app which is used to track bus movements.
 
 ### Starting the application
 
+<<<<<<< HEAD
```
Comment (samuel776) [inline README.md]: the same here remove the <<<< head are not helping you anything ie not needed

File: README.md [hunk 5]
```diff
@@ -11,7 +11,11 @@ Phantom is an app which is used to track bus movements.
...
 - Run `npm start` To start your application on `localhost:3000`.
 ### Testing
 
 - Use `npm test` to run the application's test.
+=======
+- Run `npm run dev` To start your application.
+>>>>>>> chore(react):setup React
```
Comment (samuel776) [inline README.md]: same here removes thes unwanted text of conflicts resolving

File: README.md [hunk 6]
```diff
@@ -11,7 +11,11 @@ Phantom is an app which is used to track bus movements.
 
 ### Starting the application
 
+<<<<<<< HEAD
 - Run `npm start` To start your application on `localhost:3000`.
 ### Testing
 
 - Use `npm test` to run the application's test.
+=======
```
Comment (samuel776) [inline README.md]: the same here you can remove that lines

**Review by GuevaraMarius — COMMENTED** [review]
Good work Issa, you can modify the description

File: .gitignore (hunk 1 above)
Comment (mansurissa) [reply .gitignore]: yeah

**Review by mansurissa — COMMENTED** [review]
Worked on that

**Review by GuevaraMarius — DISMISSED** [review]

File: .eslintcache [hunk 7]
```diff
@@ -0,0 +1 @@
...la/Desktop/phantom-frontend/incredible-phantom-frontend/src/componets/welcomeMessage.js":"4"},{"size":195,"mtime":1609229359713,"results":"5","hashOfConfig":"6"},{"size":291,"mtime":1609229359714,"results":"7","hashOfConfig":"6"},{"size":177,"mtime":1609229359713,"results":"8","hashOfConfig":"6"},{"size":342,"mtime":1609229359713,"results":"9","hashOfConfig":"6"},{"filePath":"10","messages":"11","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1axnbbs",{"filePath":"12","messages":"13","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"14","messages":"15","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"16","messages":"17","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/index.js",[],"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/router/App.js",[],"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/pages/landingPage.js",[],"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/componets/welcomeMessage.js",[]]
```
Comment (KundwaBruno) [inline .eslintcache]: Great work!, Please stop tracking this file `.eslintcache`

**Review by 41hulk — COMMENTED** [review]
LGTM

File: README.md (hunk 5 above)
Comment (mansurissa) [reply README.md]: worked on that

File: README.md (hunk 6 above)
Comment (mansurissa) [reply README.md]: here as well

**Review by 41hulk — DISMISSED** [review]
LGTM

File: src/index.js [hunk 8]
```diff
@@ -1,9 +1,25 @@
...
+import App from './router/App';
+import { createStore, applyMiddleware, compose } from 'redux';
+import rootReducer from './reducers/store';
+import thunk from 'redux-thunk';
+import { Provider } from 'react-redux';
+
+const store = createStore(
+  rootReducer,
+  compose(
+    applyMiddleware(thunk),
```
Comment (ceelogre) [inline src/index.js:13]: Are you using `thunk` right now or you just anticipate using it?

**Review by ceelogre — DISMISSED** [review]

**Review by KundwaBruno — DISMISSED** [review]

File: src/index.js (hunk 8 above)
Comment (mansurissa) [reply src/index.js:13]: Just unticipate using it.

**Comment by raymond42** [comment]
@mansurissa can you squash the commits please?

**Pull request closed without merging**